<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Compare Characteristic Features of Count Data Sets • countsimQC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Compare Characteristic Features of Count Data Sets">
<meta property="og:description" content="countsimQC provides functionality to create a comprehensive report 
    comparing a broad range of characteristics across a collection of count 
    matrices. One important use case is the comparison of one or more synthetic 
    count matrices to a real count matrix, possibly the one underlying the 
    simulations. However, any collection of count matrices can be compared.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">countsimQC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/countsimQC.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/csoneson/countsimQC/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="countsimqc" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#countsimqc" class="anchor"></a>countsimQC</h1></div>

<p><code>countsimQC</code> is an R package that provides functionality to create a comprehensive report comparing many different characteristics across multiple count data sets. One important use case is comparing one or more synthetic (e.g., RNA-seq) count matrices to a real count matrix, possibly the one based on which the synthetic data sets were generated. However, any collection of one or more count matrices can be visualized and compared.</p>
<p>If you use <code>countsimQC</code> for your work, we appreciate if you cite the accompanying paper:</p>
<ul>
<li>Soneson C and Robinson MD: <a href="https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btx631/4345646/Towards-unified-quality-verification-of-synthetic">Towards unified quality verification of synthetic count data with countsimQC</a>. Bioinformatics 34(4):691-692 (2018).</li>
</ul>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><code>countsimQC</code> can be installed from <a href="https://www.bioconductor.org/packages/countsimQC/">Bioconductor</a> with the following commands. Note that R version &gt;= 3.5 and Bioconductor version &gt;= 3.8 are required in order to use the <code>BiocManager</code> package. If you have an older version of R (3.4), you can still install <code>countsimQC</code> v0.5.4 (see the <code>Releases</code> tab in the GitHub repository. Please see the <code>NEWS</code> file for differences between versions.</p>
<pre><code>## Install `BiocManager` if needed
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
    
## Install countsimQC
BiocManager::install("countsimQC")</code></pre>
</div>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h2>
<p>To run <code>countsimQC</code> and generate a report, you simply need to call the function <code><a href="reference/countsimQCReport.html">countsimQCReport()</a></code>, with an input consisting of a named list of <code>DESeqDataSets</code> (see the <a href="https://www.bioconductor.org/packages/DESeq2/">DESeq2</a> package for a description of this class). Each <code>DESeqDataSet</code> should correspond to one data set and contain a count matrix, a data frame with sample information and a design formula, which is needed for proper dispersion calculations. To generate a <code>DESeqDataSet</code> from a count matrix <code>counts</code>, a sample information data frame <code>sample_df</code> and a design formula <code>formula</code> (of the form <code>~ predictors</code>), you can do as follows:</p>
<pre><code><a href="https://rdrr.io/pkg/DESeq2/man">library(DESeq2)
dds &lt;- DESeqDataSetFromMatrix(countData = counts, 
                              colData = sample_df,
                              design = formula)</a></code></pre>
<p>There are many other ways of generating valid <code>DESeqDataSets</code>, depending on in what form your counts are (e.g., reading directly from <a href="http://htseq.readthedocs.io/en/release_0.9.1/">HTSeq</a> output, or from a <a href="http://bioconductor.org/packages/release/bioc/html/tximport.html">tximport</a> output object (see the <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">vignette</a>).</p>
<p><code>countsimQC</code> contains an small example list with subsets of three data sets: two synthetic ones and the real data set that was used to generate them. The following code generates a comparative report for these three data sets:</p>
<pre><code><a href="https://rdrr.io/pkg/countsimQC/man">library(countsimQC)
data(countsimExample)
countsimQCReport(ddsList = countsimExample, 
                 outputFile = "countsimReport.html", 
                 outputDir = "./", 
                 description = "This is a comparison of three count data sets.")</a></code></pre>
<p>For more detailed information about how to use the package, we refer to the vignette:</p>
<pre><code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("countsimQC")</a></code></pre>
</div>
<div id="example-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#example-reports" class="anchor"></a>Example reports</h2>
<ul>
<li><a href="http://imlspenticton.uzh.ch/robinson_lab/countsimQC_example_reports/HMP_sampled_datasets_countsimQC.html">Comparison of 16S microbiome species count matrices for four body subsites from the Human Microbiome Project</a></li>
<li><a href="http://imlspenticton.uzh.ch/robinson_lab/countsimQC_example_reports/bulkrnaseq_crossdataset_countsimQC.html">Comparison of three real bulk RNA-seq data sets</a></li>
<li><a href="http://imlspenticton.uzh.ch/robinson_lab/countsimQC_example_reports/GSE74596_genevstx_countsimQC.html">Comparison of gene- and transcript-level count matrices for a single-cell RNA-seq data set</a></li>
<li><a href="http://imlspenticton.uzh.ch/robinson_lab/countsimQC_example_reports/scrnaseq_crossdataset_countsimQC.html">Comparison of four real scRNA-seq data sets</a></li>
<li><a href="http://imlspenticton.uzh.ch/robinson_lab/countsimQC_example_reports/GSE48968-GPL13112_2simulations_countsimQC.html">Comparison of two simulated scRNA-seq data sets to the underlying real data set</a></li>
<li><a href="http://imlspenticton.uzh.ch/robinson_lab/countsimQC_example_reports/countsimQC_compcodeR_simulations.html">Comparison of six simulated bulk RNA-seq data set with different number of genes</a></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/countsimQC">https://​www.bioconductor.org/​packages/​countsimQC</a>
</li>
<li>Browse source code at <br><a href="https://github.com/csoneson/countsimQC/">https://​github.com/​csoneson/​countsimQC/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/csoneson/countsimQC/issues">https://​github.com/​csoneson/​countsimQC/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;=2)</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing countsimQC</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Charlotte Soneson <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-3833-2169" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/csoneson/countsimQC"><img src="https://travis-ci.org/csoneson/countsimQC.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://github.com/csoneson/countsimQC/actions"><img src="https://github.com/csoneson/countsimQC/workflows/R-CMD-check/badge.svg" alt="R build status"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Charlotte Soneson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.0.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
